internal/builders/container/README.md

```diff
@@ -152,6 +153,125 @@ jobs:
+        with:
+          token_format: 'access_token'
+          workload_identity_provider: ${{ vars.WORKLOAD_IDENTITY_PROVIDER }}
+          service_account: ${{ vars.SERVICE_ACCOUNT }}
+
+      - name: Authenticate Docker
+        uses: docker/login-action@49ed152c8eca782a232dede0303416e8f356c37b # v2.0.0
+        with:
+          registry: northamerica-northeast1-docker.pkg.dev
+          username: oauth2accesstoken
+          password: ${{ steps.auth.outputs.access_token }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@69f6fc9d46f2f8bf0d5491e4aabe0bb8c6a4678a # v4.0.1
+        with:
+          images: ${{ vars.REPOSITORY_PATH }}/${{ env.IMAGE_NAME }}
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@e551b19e49efd4e98792db7592c17c09b89db8d8 # v3.0.0
+        id: build
+        with:
+          push: true
+          tags: |
+            ${{ steps.meta.outputs.tags }}
+            ${{ vars.REPOSITORY_PATH }}/${{ env.IMAGE_NAME }}:latest
+          labels: ${{ steps.meta.outputs.labels }}
+
+      - name: Output image
+        id: image
+        run: |
+          # NOTE: Set the image as an output because the `env` context is not
+          # available to the inputs of a reusable workflow call.
+          image_name="${REPOSITORY_PATH}/${IMAGE_NAME}"
+          echo image=$image_name >> "$GITHUB_OUTPUT"
+
+      - name: Output workload_identity_provider
+        id: idprov
+        run: |
+          workload_identity_provider=${WORKLOAD_IDENTITY_PROVIDER}
+          echo widp=$workload_identity_provider >> "$GITHUB_OUTPUT"
+
+      - name: Output service_account
+        id: sa
+        run: |
+          service_account=${SERVICE_ACCOUNT}
+          echo sa=$service_account >> "$GITHUB_OUTPUT"
+
+  # This step calls the container workflow to generate provenance and push it to
+  # the container registry.
+  provenance:
+    needs: [build]
+    permissions:
+      actions: read # for detecting the Github Actions environment.
+      id-token: write # for creating OIDC tokens for signing.
+      packages: write # for uploading attestations.
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_container_slsa3.yml@v1.5.0
+    with:
+      image: ${{ needs.build.outputs.image }}
+      digest: ${{ needs.build.outputs.digest }}
+      gcp-workload-identity-provider: ${{ needs.build.outputs.workload_identity_provider }}
+      gcp-service-account: ${{ needs.build.outputs.service_account }}
+      registry-username: ${{ github.actor }}
+    secrets:
+      registry-password: ${{ secrets.GITHUB_TOKEN }}
+
+```
+
+NOTE: There are existing challenges with using secrets in reusable workflows. Due to these problems, you will likely need to use unencrypted environment variables. To learn more: https://github.com/orgs/community/discussions/17554 and https://colinsalmcorner.com/consuming-environment-secrets-in-reusable-workflows/.
```

Comment:
Maybe I'm misunderstanding something but IIUC, Since the user isn't writing a reusable workflow, but rather calling one, this shouldn't really be an issue. Are you referring to something specific here?

Reply:
I'm not sure this is specific to our reusable workflows or to workload identity so maybe we can just leave this out as I think it might confuse the reader.

Or maybe you can be more specific? Like saying something like "`WORKLOAD_IDENTITY_PROVIDER` and `SERVICE_ACCOUNT` should be stored in `vars` rather than `secrets` as you cannot pass secrets to reusable workflow inputs".

wdut?